docs/tutorials/services/constrain-motion.md

```diff
@@ -0,0 +1,432 @@
+        physical_object=cup_geometry)
+]
+```
+
+Here, we use a 155mm offset along the gripper's Z axis to represent the distance from the base of the gripper to the center of the cup.
+We assume the cup is situated against the palm of the gripper's hand.
+This offset is then equal to the radius of the cup (45mm) plus the distance from the gripper's origin (where it meets the arm) to its palm (110mm in this example).
+If your gripper has different dimensions, change the offset accordingly.
+
+Now that you have created the table and tissue box obstacles and the cup transform, create a [WorldState](https://python.viam.dev/autoapi/viam/proto/common/index.html#viam.proto.common.WorldState) that includes all of them:
+
+```python {class="line-numbers linkable-line-numbers"}
+# Create a WorldState that includes the table and tissue box obstacles, and the cup transform
+world_state = WorldState(obstacles=[obstacles_in_frame], transforms = transforms)
+```
+
+## Define start and end poses
+
+You need to tell the robot where to pick up the cup, and where to put it down.
+The previous tutorial [introduced the concept of defining poses](../plan-motion-with-arm-gripper/#command-an-arm-to-move-with-the-motion-service).
+For this tutorial, you'll define a pose for the gripper that is just above the table on one side of the tissue box, and another pose on the other side of the tissue box.
+
+You'll also want to define some waypoints.
+The Motion Service has the mathematical ability to plan complex motion from one position to another, even around an obstacle.
+However, this is more computationally intensive than planning a linear path from one point to another.
+To increase your program's efficiency, add waypoints such that the path between any two consecutive points can be linear, without intersecting the tissue box or the table:
+
+```python {class="line-numbers linkable-line-numbers"}
+# Create a start pose, where the cup starts between the gripper's jaws, on the table
+# Start pose has Z of 90mm to grab partway down the 120mm tall cup
+start_pose = Pose(x=320, y=240, z=90.0+z_offset, o_x=1, o_y=0, o_z=0, theta=0)
+start_pose_in_frame = PoseInFrame(reference_frame="world", pose=start_pose)
+
+# Create waypoints to increase efficiency of motion planning
+way1_pose = Pose(x=300, y=240, z=320+z_offset, o_x=1, o_y=0, o_z=0, theta=0)
+way1_pose_in_frame = PoseInFrame(reference_frame="world", pose=way1_pose)
+way2_pose = Pose(x=300, y=-240, z=320+z_offset, o_x=1, o_y=0, o_z=0, theta=0)
+way2_pose_in_frame = PoseInFrame(reference_frame="world", pose=way2_pose)
+
+# Create a pose where the cup will be set down
+end_pose = Pose(x=300, y=-250, z=90.0+z_offset, o_x=1, o_y=0, o_z=0, theta=0)
+end_pose_in_frame = PoseInFrame(reference_frame="world", pose=end_pose)
+```
+
+Though we are passing these waypoints in manually, the Motion Service will throw an error if we accidentally pass in a pose that would cause the arm or gripper to hit any obstacles.
```

Comment:
```suggestion
Though we are passing these waypoints in manually, the Motion Service will throw an error if we accidentally pass in a pose that would cause the arm or gripper to hit any obstacles we've defined.
```

To match line 284.